File: android/src/main/kotlin/video/api/flutter/livestream/CameraSettingsHostApiImpl.kt
package video.api.flutter.livestream

import io.github.thibaultbee.streampack.utils.CameraSettings
import video.api.flutter.livestream.generated.CameraSettingsHostApi
import video.api.flutter.livestream.manager.InstanceManager

class CameraSettingsHostApiImpl(
    private val instanceManager: InstanceManager
) :
    CameraSettingsHostApi {
    private val settings: CameraSettings
        get() = instanceManager.getInstance().settings.camera

    override fun setZoomRatio(zoomRatio: Double) {
        settings.zoom.zoomRatio = zoomRatio.toFloat()
    }

    override fun getZoomRatio(): Double {
        return settings.zoom.zoomRatio.toDouble()
    }
}

File: ios/Classes/CameraSettingsHostApiImpl.swift
import AVFoundation

class CameraSettingsHostApiImpl: CameraSettingsHostApi {
    let instanceManager: InstanceManager

    init(instanceManager: InstanceManager) {
        self.instanceManager = instanceManager
    }

    func setZoomRatio(zoomRatio: Double) throws {
        guard let liveStream = instanceManager.liveStream else {
            return
        }

        liveStream.zoomRatio = CGFloat(zoomRatio)
    }

    func getZoomRatio() throws -> Double {
        guard let liveStream = instanceManager.liveStream
        else {
            return 1.0
        }
        return Double(liveStream.zoomRatio)
    }
}
